codec-http/src/main/java/io/netty/handler/codec/http/cookie/CookieHeaderNames.java

```diff
@@ -28,6 +28,35 @@
 
     public static final String HTTPONLY = "HTTPOnly";
 
+    public static final String SAMESITE = "SameSite";
+
+    /**
+     * Possible values for the SameSite attribute.
+     * See <a href="https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-05">changes to RFC6265bis</a>
+     */
+    public enum SameSite {
+        Lax,
+        Strict,
+        None;
+
+        /**
+         * Return the enum value corresponding to the passed in same-site-flag, using a case insensitive comparison.
+         *
+         * @param name value for the SameSite Attribute
+         * @return enum value for the provided name or null
+         */
+        static SameSite of(String name) {
+            if (name != null) {
+                for (SameSite each : SameSite.class.getEnumConstants()) {
+                    if (each.name().equalsIgnoreCase(name)) {
+                        return each;
+                    }
+                }
+            }
+            return null;
```

Comment:
should be non null, according to spec it should fallback to None..

https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-05#section-4.1

If the "SameSite" attribute's value is "Strict", the cookie will only
   be sent along with "same-site" requests.  If the value is "Lax", the
   cookie will be sent with same-site requests, and with "cross-site"
   top-level navigations, as described in Section 5.3.7.1.  If the value
   is "None", the cookie will be sent with same-site and cross-site
   requests.  **If the "SameSite" attribute's value is something other
   than these three known keywords, the attribute's value will be
   treated as "None".**

Reply:
I understand "will be treated" to refer to the behaviour explained about regarding whether a cookie will be sent or not:  
 ```
If the value is "None", the cookie will be sent with same-site and cross-site
   requests
```

Both here and in the normative section (https://tools.ietf.org/html/draft-ietf-httpbis-rfc6265bis-05#section-5.3.7) it mentions behaviour or "treated as" and in no case it seems to mention we should replace the value on the wire. 

That section might mean we should always keep the received value for `SameSite` but we have agreed on the contrary. In any case, cookies are most likely going to be revisited in the next major version - I would suggest that we could keep unknown attributes as well and not only unknown values so things like this "new" SameSite attribute don't  break netty.